docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,186 @@
+---
+category: framework-deep-dive
+---
+
+# Localization
+
+## Introduction
+
+The CKEditor 5 ecosystem support message localization. It means that the UI of any feature can be translated for various languages and regions depending on the user's preferences. With the the recent release of the CKEditor 5 packages and CKEditor 5 dev tools we opened the translation system for 3rd-party plugins, provided a way for adding missing or fixing invalid translations to the editor and we introduced support for translating plural forms.
+
+### Open API
+
+Until now the building process was creating short strings for *message IDs*. This behavior was confusing, it caused non-deterministic builds (as the ID was generated during the webpack compilation) and made adding translations to the build impossible as IDs have been changing with each new CKEditor 5 release and with each project rebuilding. So we decided to drop the *message ID* shortening feature, as the gain from lower bundle size was disproportionately lower than the loss from the mentioned issues. That means, that from now:
```

Comment:
Here's something that I'd like to emphasize:

We write the docs for developers outside _CKEditor organization_, basically any developer out there willing to play with the editor.

This documentation should focus mostly on how to do stuff, maybe a little why some things are done the way they are but it should not explain how stuff was done in the past unless it's really necessary.

Here we don't really need to explain how things were handled in the past (especially that they were not available for anyone outside CKEditor organization).

Reply:
Good point :+1:

Reply:
I changed the whole chapter.